### PR TITLE
Fix/user apps

### PR DESCRIPTION
**Pull request by VelvetToroyashi**

### CODE REVIEW

**Comment by Nihlus** [comment]
You're still missing test payloads for the added entities and tests for those.

**Comment by VelvetToroyashi** [comment]
For interaction objects? Or the application object

**Comment by Nihlus** [comment]
The JSON payloads under test haven't been updated for existing entities (IApplicationCommand) and no new JSON payloads have been added for new types (like IMessageInteractionMetadata). Check your changed and added entities from both this PR and #327.

**Review by MazeXP — CHANGES_REQUESTED** [review]
LGTM, just some small things. :)

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Interactions/IMessageInteractionMetadata.cs [hunk 1]
```diff
@@ -40,7 +40,12 @@ public interface IMessageInteractionMetadata
     /// <summary>
     /// Gets the ID of the user who triggered the interaction.
     /// </summary>
-    Snowflake UserID { get; }
+    IUser User { get; }
+
+    /// <summary>
+    /// Gets the name of the command.
+    /// </summary>
+    Optional<string> Name { get; }
```
[the remark this answers is not in the document]
Comment (VelvetToroyashi) [reply Backend/Remora.Discord.API.Abstractions/API/Objects/Interactions/IMessageInteractionMetadata.cs:48]: Was added between the PR being opened, and Shay marking it as unreleased https://github.com/discord/discord-api-docs/pull/6734

File: Backend/Remora.Discord.API/Json/Converters/Internal/EnumKeyDictionaryConverterFactory.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Remora.Rest.Json;
+
+namespace Remora.Discord.API.Json;
+
+/// <summary>
+/// Creates a JSON converter for dictionaries with enum keys.
+/// </summary>
+public class EnumKeyDictionaryConverterFactory : JsonConverterFactory
```
[the remark this answers is not in the document]
Comment (VelvetToroyashi) [reply Backend/Remora.Discord.API/Json/Converters/Internal/EnumKeyDictionaryConverterFactory.cs]: The semantics on this class are a little scuffed; technically a stringified integer, so perhaps it'd be better to clarify in the documentation?

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Interactions/IMessageInteractionMetadata.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MazeXP) [reply Backend/Remora.Discord.API.Abstractions/API/Objects/Interactions/IMessageInteractionMetadata.cs:48]: Alright, Jax must decide on it

File: Backend/Remora.Discord.API/Json/Converters/Internal/EnumKeyDictionaryConverterFactory.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MazeXP) [reply Backend/Remora.Discord.API/Json/Converters/Internal/EnumKeyDictionaryConverterFactory.cs]: Well, in JSON terms a key of an object is always a string

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Interactions/IMessageInteractionMetadata.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nihlus) [reply Backend/Remora.Discord.API.Abstractions/API/Objects/Interactions/IMessageInteractionMetadata.cs:48]: If it's not documented yet, don't implement it. Especially not when Discord themselves say "this is unstable and intentionally not documented".

**Comment by Nihlus** [comment]
You've got a few alerts remaining along with some merge conflicts that came in as part of polls - other than that, things seem ready to go.

**Comment by VelvetToroyashi** [comment]
@Nihlus should be good to go now

**Pull request merged**